aiida_nanotech_empa/workflows/gaussian/delta_scf_workchain.py

```diff
@@ -0,0 +1,244 @@
+from aiida_nanotech_empa.workflows.gaussian import common
+
+from aiida.engine import WorkChain, calcfunction, ExitCode
+from aiida.orm import Int, Str, Code, Dict
+from aiida.orm import StructureData, RemoteData
+
+from aiida.plugins import WorkflowFactory
+
+GaussianBaseWorkChain = WorkflowFactory('gaussian.base')
+
+
+@calcfunction
+def subtract(a, b):
+    return a - b
+
+
+class GaussianDeltaScfWorkChain(WorkChain):
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+
+        spec.input("gaussian_code", valid_type=Code)
+
+        spec.input('structure',
+                   valid_type=StructureData,
+                   required=True,
+                   help='input geometry')
+        spec.input('functional',
+                   valid_type=Str,
+                   required=True,
+                   help='xc functional')
+
+        spec.input('basis_set',
+                   valid_type=Str,
+                   required=True,
+                   help='basis_set')
+
+        spec.input('multiplicity',
+                   valid_type=Int,
+                   required=False,
+                   default=lambda: Int(1),
+                   help='spin multiplicity; 0 means RKS')
+
+        spec.input(
+            "parent_calc_folder",
+            valid_type=RemoteData,
+            required=False,
+            help="the folder of a completed gaussian calculation",
+        )
+
+        spec.outline(cls.setup, cls.submit_scfs, cls.finalize)
+
+        spec.outputs.dynamic = True
+
+        spec.exit_code(
+            301,
+            "ERROR_MULTIPLICITY",
+            message="Multiplicity and number of el. doesn't match.",
+        )
+        spec.exit_code(
+            390,
+            "ERROR_TERMINATION",
+            message="One or more steps of the work chain failed.",
+        )
+
+    def is_uks(self):
+        if self.inputs.multiplicity.value == 0:
+            return False
+        return True
+
+    def setup(self):
+        self.report("Inspecting input and setting up things.")
+
+        common.setup_context_variables(self)
+
+        if self.ctx.mult % 2 == self.ctx.n_electrons % 2:
+            return self.exit_codes.ERROR_MULTIPLICITY
+
+        return ExitCode(0)
+
+    def setup_common_builder_params(self, builder):
+        builder.gaussian.structure = self.inputs.structure
+        builder.gaussian.code = self.inputs.gaussian_code
+        common.set_metadata(builder.gaussian.metadata, self.ctx.n_atoms,
+                            self.ctx.comp)
```

Comment:
I think doing this in the work chain is quite a bad idea. Can you not determine the resources a priory and inject them during the submission process? This comment goes for all the other work chains.

This function should be used as a suggestion, not as "the solution", because there are plenty of corner cases with this approach.

Reply:
Done. now you can input `options` to the workchains (and if it's not inputted, then the automatic determination is done). I used the [aiida-quantumespresso band structure](https://github.com/aiidateam/aiida-quantumespresso/blob/2c6d02f69cf273cb5f78f6f9f39839d6397793ae/aiida_quantumespresso/workflows/pw/band_structure.py#L137-L140) as an example.